### PR TITLE
Fix/test outbox put message

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

addresses: https://github.com/valory-xyz/open-aea/issues/349

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`aea hash all` and `aea packages lock --check`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: tests/test_mail/test_base.py [hunk 1]
```diff
@@ -210,12 +215,16 @@ def test_outbox_put_message():
     inbox = InBox(multiplexer)
     multiplexer.connect()
     wait_for_condition(
-        lambda: multiplexer.is_connected, 30, "Multiplexer is not connected"
+        lambda: multiplexer.is_connected, TIMEOUT, "Multiplexer is not connected"
     )
     outbox.put_message(msg)
     wait_for_condition(
-        lambda: inbox.empty(), 30, "Inbox must not be empty after putting a message"
+        lambda: not inbox.empty(),
```
Comment (Karrenbelt) [inline tests/test_mail/test_base.py:222]: this was the issue

File: tests/test_mail/test_base.py [hunk 2]
```diff
@@ -186,7 +189,9 @@ def test_outbox_put():
     )
     outbox.put(envelope)
     wait_for_condition(
-        lambda: inbox.empty(), 15, "Inbox must not be empty after putting an envelope"
+        lambda: inbox.empty(),
+        TIMEOUT,
+        "Inbox must not be empty after putting an envelope",
```
Comment (Adamantios) [inline tests/test_mail/test_base.py:194]: Isn't this a similar case?

Comment (Karrenbelt) [reply tests/test_mail/test_base.py:194]: lol I didn't notice. It's good that I did the refactor to introduce the `TIMEOUT` constant so you would notice this. Credits to you though for catching this!

**Comment by Karrenbelt** [comment]
failures seen:

```
# Ubuntu 3.8, 3.9, MacOS 3.9, Win 3.8, 3.9, 3.10: 
# https://github.com/valory-xyz/open-aea/actions/runs/3220440624/jobs/5269756693
FAILED tests/test_cli/test_utils/test_utils.py::GetOrCreateCLIConfigTestCase::testget_or_create_cli_config_positive
 >       self.assertEqual(result, expected_result)
E       AssertionError: {'author': 'default_author', 'registry_conf[174 chars]e}}}} != {'correct': 'output'}
E       + {'correct': 'output'}
E       - {'author': 'default_author',
E       -  'registry_config': {'default': 'local',
E       -                      'settings': {'local': {'default_packages_path': None},
E       -                                   'remote': {'http': {'auth_token': None,
E       -                                                       'registry_api_url': None},
E       -                                              'ipfs': {'ipfs_node': None}}}}}

FAILED tests/test_cli/test_utils/test_utils.py::GetOrCreateCLIConfigTestCase::testget_or_create_cli_config_bad_yaml
        with self.assertRaises(ClickException):
>           get_or_create_cli_config()
E           AssertionError: ClickException not raised

FAILED tests/test_cli/test_utils/test_utils.py::InitConfigFolderTestCase::test_init_cli_config_positive
>           get_or_create_cli_config()
>       _init_cli_config(user_config)
E       FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpryn4v32h/cli_config.yaml'


# MacOS 3.10: 
# https://github.com/valory-xyz/open-aea/actions/runs/3220440624/jobs/5269757083
FAILED tests/test_contracts/test_base.py::test_get_deploy_transaction_cosmwasm
FAILED tests/test_decision_maker/test_default.py::TestDecisionMaker::test_handle_tx_signing_fetchai
>       transfer_transaction = fetchai_api.get_transfer_transaction(
>       raise TypeError(
            'Parameter to MergeFrom() must be instance of same class: '
            'expected %s got %s.' % (_FullyQualifiedClassName(cls),
                                     _FullyQualifiedClassName(msg.__class__)))
E       TypeError: Parameter to MergeFrom() must be instance of same class: expected cosmos.base.v1beta1.coin_pb2.Coin got cosmos.base.v1beta1.coin_pb2.Coin.
```

**Review by Adamantios — APPROVED** [review]

**Comment by angrybayblade** [comment]
`testget_or_create_cli_config_bad_yaml` looks flaky, issue for the same https://github.com/valory-xyz/open-aea/issues/366

`E       TypeError: Parameter to MergeFrom() must be instance of same class: expected cosmos.base.v1beta1.coin_pb2.Coin got cosmos.base.v1beta1.coin_pb2.Coin.` Looks like a fetchai's testnet related failure, we've had one of these before.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**